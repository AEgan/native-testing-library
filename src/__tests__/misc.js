import React from 'react';
import { Picker } from 'react-native';

import { cleanup, fireEvent, render } from '../';

afterEach(cleanup);

test('picker works', () => {
  function Wrapper() {
    const [value, setValue] = React.useState('js');

    return (
      <Picker selectedValue={value} onValueChange={itemValue => setValue(itemValue)}>
        <Picker.Item label="Java" value="java" />
        <Picker.Item label="JavaScript" value="js" />
      </Picker>
    );
  }
  const { findByDisplayValue, getByDisplayValue } = render(<Wrapper />);

  fireEvent.valueChange(getByDisplayValue('js'), 'java');
  expect(() => findByDisplayValue('js')).not.toThrow();
});
