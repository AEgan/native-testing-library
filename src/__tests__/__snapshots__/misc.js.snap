// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fragments can show diffs 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -6,8 +6,8 @@
        \\"flex\\": 1,
      }
    }
  >
    <Button
-     title=\\"Click to increase: 0\\"
+     title=\\"Click to increase: 1\\"
    />
  </View>"
`;
